feat(linter): allow `eqeqeq` to always be dangerously fixable

camchenry opened the pull request:
- Related to https://github.com/oxc-project/oxc/issues/10477

This allows the `eqeqeq` rule to always be fixable if dangerous fixes are allowed, but will apply the safe fix if the circumstances apply.

camchenry commented:
* **#10499** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10499?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10499?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10499).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on crates/oxc_linter/src/rules/eslint/eqeqeq.rs:
```diff
@@ -88,7 +88,7 @@ declare_oxc_lint!(
     eslint,
     pedantic,
-    conditional_fix
+    conditional_fix_dangerous,
```
I am not sure how the fixer internally works. What happens when we use only safe fixes? Does the Fixer prefilter them and ignore the safe fixes?
Maybe we want to create some test for it?

camchenry replied:
I'm not entirely sure what you mean: as far as this line goes, this is declaring that the rule _can_ emit dangerous fixes, but it is limited to _only_ emitting dangerous fixes. The safeness is per-fix, so even with just passing `oxlint --fix`, this will still fix safe `eqeqeq` violations, but you will need to pass `oxlint --fix --fix-dangerously` to fix all.

Sysix replied:
> The safeness is per-fix, so even with just passing oxlint --fix, this will still fix safe eqeqeq violations, but you will need to pass oxlint --fix --fix-dangerously to fix all.

That is the expected behavior, I am not sure if that is working as intended :)

camchenry commented:
### Merge activity

* **Apr 19, 12:52 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
feat(linter): allow `eqeqeq` to always be dangerously fixable (#10499)

- Related to https://github.com/oxc-project/oxc/issues/10477

This allows the `eqeqeq` rule to always be fixable if dangerous fixes are allowed, but will apply the safe fix if the circumstances apply.
